cpp/sammelschrift_statistik.cc

```diff
@@ -0,0 +1,236 @@
+        if (not a_contents.empty()) {
+            if (a_contents == "Aufsatzsammlung" or a_contents == "Festschrift" or a_contents == "Konferenzschrift")
+                return true;
+        }
+    }
+
+    return false;
+}
+
+
+std::string GetShortenedTitle(const MARC::Record &record, const size_t max_length) {
+    auto complete_title(record.getCompleteTitle());
+    if (complete_title.length() > max_length) {
+        if (unlikely(not TextUtil::UnicodeTruncate(&complete_title, max_length)))
+            LOG_ERROR("bad Unicode in title of record with PPN " + record.getControlNumber() + "!");
+        return complete_title + "...";
+    }
+
+    return complete_title;
+}
+
+
+bool IsPossibleYear(const std::string &year_candidate) {
+    if (year_candidate.length() != 4)
+        return false;
+
+    for (auto ch : year_candidate) {
+        if (not StringUtil::IsDigit(ch))
+            return false;
+    }
+
+    return true;
+}
+
+
+std::string YYMMDateToString(const std::string &control_number, const std::string &yymm_date) {
+    const unsigned CURRENT_YEAR(TimeUtil::GetCurrentTimeGMT().tm_year + 1900);
+    const unsigned TWO_DIGIT_CURRENT_YEAR(CURRENT_YEAR - 2000);
+
+    unsigned year_digits;
+    if (StringUtil::ToUnsigned(yymm_date.substr(0, 2), &year_digits))
+        return std::to_string(year_digits > TWO_DIGIT_CURRENT_YEAR ? 1900 + year_digits : 2000 + year_digits);
+    LOG_WARNING("in yyMMDateToString: expected date in YYMM format, found \"" + yymm_date
+                + "\" instead! (Control number was " + control_number + ")");
+    return std::to_string(CURRENT_YEAR);
+}
+
+
+std::string GetDateForWebsite(const MARC::Record &record) {
+    const auto _008_field(record.findTag("008"));
+    const auto &control_number(record.getControlNumber());
+    if (unlikely(_008_field == record.end()))
+        LOG_ERROR("No 008 Field for website w/ control number " +  control_number + "!");
+    return YYMMDateToString(control_number, _008_field->getContents());
+}
+
+
+std::string GetDateForReproduction(const MARC::Record &record) {
+    const auto _534_field(record.findTag("534"));
+    const auto &control_number(record.getControlNumber());
+    if (unlikely(_534_field == record.end()))
+        LOG_ERROR("No 534 Field for reproduction w/ control number " +  control_number + "!");
+
+    const auto c_contents(_534_field->getFirstSubfieldWithCode('c'));
+    if (c_contents.empty())
+        return "";
+
+    static const auto digit_matcher(RegexMatcher::RegexMatcherFactoryOrDie("(\\d+)"));
+    return digit_matcher->matched(c_contents) ? (*digit_matcher)[1] : "";
+}
+
+
+std::string GetDateForArticleOrReview(const MARC::Record &record) {
+    for (const auto &_936_field : record.getTagRange("936")) {
+        const auto j_contents(_936_field.getFirstSubfieldWithCode('j'));
+        if (not j_contents.empty()) {
+            static const auto year_matcher(RegexMatcher::RegexMatcherFactoryOrDie("\\d{4}"));
```

Comment:
Also if seen strictly, not the same behaviour as getDatesBasedOnRecordType (side effects?)